### PR TITLE
add: support for different currencies in dashboard

### PR DESCRIPTION
**Pull request by chdsbd**
Add support for giving a user a different pricing plan. For example, a user may require that we bill in EUR instead of USD. We can create a pricing plan in Stripe in EUR and configure the dashboard to use that for billing instead of the default plan_id.

Add support for display different currencies in web ui.

image

image

TODO in future PR
- display plan-specific pricing in subscription creation form

### CODE REVIEW

File: web_ui/src/components/UsageBillingPage.tsx [hunk 1]
```diff
@@ -662,6 +686,10 @@ function ActiveSubscription({
   modifySubscription,
   teamId,
 }: IActiveSubscriptionProps) {
+  const formatter = new Intl.NumberFormat("en-US", {
```
Comment (sbdchd) [inline web_ui/src/components/UsageBillingPage.tsx:689]:
```suggestion
  const formatter = new Intl.NumberFormat(undefined, {
```

File: web_ui/src/components/UsageBillingPage.tsx [hunk 2]
```diff
@@ -110,8 +113,12 @@ function UsageAndBillingContainer({ children }: { children: React.ReactNode }) {
   )
 }
 
-function formatCents(cents: number): string {
-  return `\$${(cents / 100).toFixed(2)}`
+function formatCents(cents: number, currency: string): string {
```
Comment (sbdchd) [inline web_ui/src/components/UsageBillingPage.tsx:116]: why aren't we using this function down below?

File: web_ui/src/components/UsageBillingPage.tsx [hunk 3]
```diff
@@ -691,9 +719,11 @@ function ActiveSubscription({
           <b>Cost</b>
         </Col>
         <Col>
-          <span className="mr-4">{formatCents(cost.totalCents)} / month</span>
+          <span className="mr-4">
+            {formatter.format(cost.perSeatCents / 100)} / month
+          </span>
           <span>
-            ({formatCents(cost.perSeatCents)} / seat ⨉ {seats} seats)
+            ({formatter.format(cost.perSeatCents / 100)} / seat ⨉ {seats} seats)
```
Comment (sbdchd) [inline web_ui/src/components/UsageBillingPage.tsx:726]: like here?

File: web_ui/src/components/UsageBillingPage.tsx (hunk 2 above)
Comment (chdsbd) [reply web_ui/src/components/UsageBillingPage.tsx:116]: fixed

**Pull request merged**